controllers/externalservice_controller_test.go

```diff
@@ -145,7 +146,7 @@ func assertState(key types.NamespacedName, es *v1.ExternalService) {
 
 		return d
 	}, timeout, interval).Should(And(
-		WithTransform(func(d *appsv1.Deployment) appsv1.DeploymentSpec { return d.Spec }, Equal(deployment(es, cHash).Spec)),
+		WithTransform(func(d *appsv1.Deployment) appsv1.DeploymentSpec { return d.Spec }, BeComparableTo(deployment(es, cHash).Spec)),
```

Comment:
BeComparableTo is like Equal except it uses go-cmp under the hood instead of DeepCopy - this makes it much easier to see the actual object diff